src/replication/ng/scheduler/scheduler.go

```diff
@@ -42,3 +64,85 @@ type Scheduler interface {
 	// Stop the job specified by ID
 	Stop(id string) error
 }
+
+// Preprocess the resources and returns the item list that can be scheduled
+func (d *DefaultReplicator) Preprocess(srcResources []*model.Resource, destResources []*model.Resource) ([]*ScheduleItem, error) {
+	if len(srcResources) != len(destResources) {
+		err := errors.New("srcResources has different length with destResources")
+		log.Errorf(err.Error())
+		return nil, err
+	}
+	var items []*ScheduleItem
+	for index, srcResource := range srcResources {
+		destResource := destResources[index]
+		item := &ScheduleItem{
+			SrcResource: srcResource,
+			DstResource: destResource,
+		}
+		items = append(items, item)
+
+	}
+	return items, nil
+}
+
+// Schedule transfer the tasks to jobs,and then submit these jobs to job service.
+func (d *DefaultReplicator) Schedule(items []*ScheduleItem) ([]*ScheduleResult, error) {
+	var results []*ScheduleResult
+	for _, item := range items {
+		if item.TaskID == 0 {
+			err := errors.New("some tasks do not have a ID")
+			log.Errorf(err.Error())
+			return nil, err
```

Comment:
Put the error to its result, don't mark the whole function failed, or the flow controller will mark the whole execution failed.